k8s/pkg/prefixcollector/collector_server_test.go

```diff
@@ -0,0 +1,142 @@
+package prefixcollector
+
+import (
+	"fmt"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"net"
+	"testing"
+	"time"
+)
+
+type dummyResource struct {
+	name   string
+	subnet string
+}
+
+func (*dummyResource) GetObjectKind() schema.ObjectKind {
+	panic("implement me")
+}
+
+func (*dummyResource) DeepCopyObject() runtime.Object {
+	panic("implement me")
+}
+
+func keyFuncDummy(event watch.Event) (string, error) {
+	return event.Object.(*dummyResource).name, nil
+}
+
+func subnetFuncDummy(event watch.Event) (*net.IPNet, error) {
+	_, ipNet, _ := net.ParseCIDR(event.Object.(*dummyResource).subnet)
+	return ipNet, nil
+}
+
+type dummyWatcher struct {
+	eventCh chan watch.Event
+}
+
+func NewDummyWatcher() *dummyWatcher {
+	return &dummyWatcher{
+		eventCh: make(chan watch.Event),
+	}
+}
+
+func (d *dummyWatcher) Stop() {
+	panic("implement me")
+}
+
+func (d *dummyWatcher) ResultChan() <-chan watch.Event {
+	return d.eventCh
+}
+
+func (d *dummyWatcher) send(t watch.EventType, dr *dummyResource) {
+	d.eventCh <- watch.Event{
+		Type:   t,
+		Object: dr,
+	}
+}
+
+func checkSubnetWatcher(t *testing.T, subnetSequence, expectedSequence []string) {
+	dw := NewDummyWatcher()
+	sw, err := watchSubnet(dw, keyFuncDummy, subnetFuncDummy)
+	Expect(err).To(BeNil())
+
+	for i := 0; i < len(subnetSequence); i++ {
+		dw.send(watch.Added, &dummyResource{fmt.Sprintf("r%d", i), subnetSequence[i]})
+	}
+
+	for i := 0; i < len(expectedSequence); i++ {
+		select {
+		case e := <-sw.ResultChan():
+			Expect(e.String()).To(Equal(expectedSequence[i]))
+		case <-time.Tick(1 * time.Second):
+			if expectedSequence[i] != "-" {
+				logrus.Error("Timeout waiting for next subnet")
+				t.Fail()
+			}
+		}
+	}
+}
+
+func TestSimpleSubnetCollector(t *testing.T) {
+	RegisterTestingT(t)
+	subnetSequence := []string{
+		"10.20.1.0/24",
+		"10.20.2.0/24",
+	}
+	expectedSubnets := []string{
+		"10.20.1.0/24",
+		"10.20.0.0/22",
+	}
+	checkSubnetWatcher(t, subnetSequence, expectedSubnets)
+}
+
+func TestLastIpsAlreadyInSubnet(t *testing.T) {
```

Comment:
Please keep the tests in a seaprate file.

Reply:
What do you mean by separate file? Should I split `dummyWatcher` implementation and tests?

Reply:
It is already a test file, no need to change anything.